File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> stk;
        for(int i = 0; i < tokens.size(); i++){
            if(tokens[i] != "+" && tokens[i] != "-" &&
             tokens[i] != "*" && tokens[i] != "/"){
                int value = stoi(tokens[i]);
                stk.push(value);
            }
            else{
                char cur = tokens[i][0];
                int val1,val2,result;
                if(!stk.empty()){
                        val1 = stk.top();
                        stk.pop();
                }
                else{
                    return 0;
                }
                if(!stk.empty()){
                    val2 = stk.top();
                    stk.pop();
                }
                else{
                    return 0;
                }
                if(cur == '+'){
                    result = val1 + val2;
                }
                else if(cur == '-'){
                    result = val2 - val1;
                }
                else if(cur == '*'){
                    result = val1 * val2;
                }
                else if(cur == '/'){
                    result = val2 / val1;
                }
                stk.push(result);
            }
        }
        if(stk.empty()){
            return 0;
        }
        return stk.top();
    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-2.cs
public class Solution {
    public int[] DailyTemperatures(int[] temperatures) {
        Stack<(int value, int index, int counter)> stack = new Stack<(int value, int index, int counter)>();

        int[] result = new int[temperatures.Length];

        (int value, int index, int counter) top = (temperatures[0], 0, 0);
        stack.Push(top);
        int index = 1;
        while(stack.Count != 0 && index < temperatures.Length){
            top = stack.Peek();
            int counter = 0;
            int deletedTillNow = top.counter;
            while(stack.Count != 0 && top.value < temperatures[index]){
                counter++;
                stack.Pop();
                result[top.index] = deletedTillNow - top.counter + counter;
                if(stack.Count != 0)
                    top = stack.Peek();
                }
            stack.Push((temperatures[index], index, deletedTillNow + counter));
            index += 1;
        }
        return result;
    }
}
